clang/lib/Sema/SemaStmt.cpp

```diff
@@ -4556,8 +4556,13 @@ EqualityOpFact *Sema::ActOnEqualityOpFact(Expr *E, SourceLocation ExprLoc) {
 
   auto *BO = dyn_cast_or_null<BinaryOperator>(TmpE);
 
+  if (!BO) {
```

Comment:
In addition to the condition `!BO`, it will be great if we can ensure that we are in a WhereClause scope here because the error message is only relevant in the context of the where clause.

Reply:
This function (`ActOnEqualityOpFact`) is only called when we are in a `WhereClauseScope` (ie: while parsing a where clause). So I am not sure how explicitly checking for `WhereClauseScope` would help?

Reply:
It may be a good idea to keep ActOnEqualityOpFact general - it may be called (in the future) on "inferred facts" which may not always be embedded in a Where clause.